subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -323,13 +323,13 @@ static int aws_iot_topics_populate(char *const id, size_t id_len)
 #if defined(CONFIG_AWS_IOT_CLIENT_ID_APP)
 	err = snprintf(client_id_buf, sizeof(client_id_buf),
 		       AWS_CLIENT_ID_PREFIX, id);
-	if (err >= AWS_CLIENT_ID_LEN_MAX) {
+	if (err > AWS_CLIENT_ID_LEN_MAX) {
```

Comment:
That's why I personally prefer to use the actual buffer size in such checks (`err => sizeof(client_id_buf)`). But we can keep it this way if you prefer.

Reply:
Perhaps change the it to `if (err < 0 || err >= sizeof(client_id_buf)`  to catch errors returned by `snprintf` and make sure the check also works if `AWS_CLIENT_ID_PREFIX` changes

Reply:
> `err >= sizeof(client_id_buf))`

Why `>=` and not `>`? The return code of `snprintf` is the number of characters written without the trailing null character. So if it is the same length as the buffer the string should not be too long?

Reply:
`snprintf` always includes a trailing `\0`, so if the return value is equal to the buffer size, the last character has been truncated

Reply:
Ah, I get it. This is taken care of here:
https://github.com/nrfconnect/sdk-nrf/blob/9876fb8ca129794b42c68425cd3202181c7189ff/subsys/net/lib/aws_iot/src/aws_iot.c#L44

Reply:
> Perhaps change the check to `if (err < 0 || err >= sizeof(client_id_buf))` to catch errors returned by `snprintf` and make sure the check also works if `AWS_CLIENT_ID_PREFIX` changes

Use this check instead.